scripts/quests/crystalWar/The_Lost_Book.lua

```diff
@@ -0,0 +1,203 @@
+-----------------------------------
+-- The Lost Book
+-- !addquest 7 26
+--
+-- MYTHRIL_BEASTCOIN - !additem 749
+-- VELLUM - !additem 2550
+--
+-- Rhinostery Door South - !pos -1 -4.00 -198 94
+-- Quu Bokye - !pos -159 16 181 145
+-- Optistery Door - !pos -57 -5.0 89 94
+-- qm0 - !pos -141 1 -9 99
+-----------------------------------
+require('scripts/globals/interaction/quest')
+require('scripts/globals/npc_util')
+require('scripts/globals/keyitems')
+require('scripts/globals/quests')
+require('scripts/globals/items')
+require('scripts/globals/zone')
+-----------------------------------
+local castleOID = require("scripts/zones/Castle_Oztroja_[S]/IDs")
+local giddeusID = require("scripts/zones/Giddeus/IDs")
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.CRYSTAL_WAR, xi.quest.id.crystalWar.THE_LOST_BOOK)
+
+quest.reward =
+{
+    item = xi.items.SCROLL_OF_RETRACE,
+}
+
+quest.sections =
+{
+    -- Examine the right Rhinostery door (J-9 of the second map) for a cutscene.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and player:getMainLvl() >= 30 and
+                player:hasKeyItem(xi.ki.BRONZE_RIBBON_OF_SERVICE)
+        end,
+
+        [xi.zone.WINDURST_WATERS_S] =
+        {
+            ['Door_Rhinostery_South'] = quest:progressEvent(143),
+
+            onEventFinish =
+            {
+                [143] = function(player, csid, option, npc)
+                    quest:begin(player)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+            },
+        },
+    },
+
+    -- Bring a Mythril Beastcoin and travel to present-day Giddeus and trade Quu Bokye
+    -- a Mythril Beastcoin to obtain a Leather-bound Book.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 1
+        end,
+
+        [xi.zone.WINDURST_WATERS_S] =
+        {
+            ['Door_Rhinostery_South'] = quest:progressEvent(148), -- Optional dialogue
+        },
+
+        [xi.zone.GIDDEUS] =
+        {
+            ['Quu_Bokye'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, xi.items.MYTHRIL_BEASTCOIN) then
+                        return quest:progressEvent(65)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    return quest:progressEvent(64)
+                end,
+            },
+
+            onEventFinish =
+            {
+                [65] = function(player, csid, option, npc)
+                    player:confirmTrade()
+                    npcUtil.giveKeyItem(player, xi.ki.LEATHERBOUND_BOOK)
+                    quest:messageSpecial(giddeusID.text.KEYITEM_OBTAINED, xi.ki.LEATHERBOUND_BOOK)
```

Comment:
I thought npcUtil handled sending this message?

Reply:
Same here. Tested it twice and it would not print it. Check another converted quest and saw messageSpecial used still, so threw it in there.

Reply:
Weird, because npcUtil should be showing this message.